x-pack/plugin/sql/qa/jdbc/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/JdbcTestUtils.java

```diff
@@ -5,25 +5,76 @@
  */
 package org.elasticsearch.xpack.sql.qa.jdbc;
 
+import org.elasticsearch.Version;
+import org.elasticsearch.xpack.sql.jdbc.EsType;
 import org.elasticsearch.xpack.sql.proto.StringUtils;
 
+import java.math.BigInteger;
 import java.sql.Date;
 import java.sql.Time;
+import java.sql.Timestamp;
 import java.time.Instant;
 import java.time.LocalDate;
+import java.time.LocalDateTime;
 import java.time.ZoneId;
 import java.time.ZoneOffset;
 import java.time.ZonedDateTime;
 import java.util.Calendar;
+import java.util.Collections;
+import java.util.GregorianCalendar;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import static org.elasticsearch.Version.V_7_11_0;
 
 final class JdbcTestUtils {
 
     private JdbcTestUtils() {}
 
+    private static final Map<Class<?>, EsType> CLASS_TO_ES_TYPE;
+
     static final ZoneId UTC = ZoneId.of("Z");
     static final String JDBC_TIMEZONE = "timezone";
+    private static final String DRIVER_VERSION_PROPERTY_NAME = "jdbc.driver.version";
     static final LocalDate EPOCH = LocalDate.of(1970, 1, 1);
 
+    static final Version JDBC_DRIVER_VERSION;
+
+    static {
```

Comment:
A similar code is done inside the actual driver. Can we reuse it somehow so we don't have multiple mapping initializations that can (and eventually will) get out of sync? If SysTypes is versioned, does it make sense to initialize this map dynamically by calling the server API?
Or at least having a comment in *both* places so the code gets updated at once.

Reply:
I've added comments.

Initialising a list with supported types is def. doable, but mapping those to Java classes would still require a "manual" map. Alternatively, we could extend `EsType` to include the class?